pkg/utils/validation/apigroups/apigroups.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apigroups
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	versionutils "github.com/gardener/gardener/pkg/utils/version"
+)
+
+var (
+	// apiGroupVersionRanges contains the version ranges for all Kubernetes API GroupVersions.
+	// Extracted from https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/staging/src/k8s.io/client-go/informers/generic.go
+	// To maintain this list for each new Kubernetes version:
+	// - To maintain this list for new Kubernetes versions, run `hack/compare-k8s-api-groups.sh <old-version> <new-version>` (e.g. `hack/compare-k8s-api-groups.sh 1.26 1.27`).
+	// - It will present 2 lists of API GroupVersions and 2 lists of API GroupVersionResources: those added and those removed in `<new-version>` compared to `<old-version>`.
+	// - Add all added group versions to the `apiGroupVersionRanges` map and group version resources to the `apiGVRVersionRanges` map with `<new-version>` as `AddedInVersion` and no `RemovedInVersion`.
+	// - For any removed APIs, add `<new-version>` as `RemovedInVersion` to the already existing API in the corresponding map.
+	// - Flag any APIs that are required (APIs that must not be disabled in the `Shoot` spec) by setting the `Required` and `RequiredForWorkerless` boolean variable to true for the API in the `apiGVRVersionRanges` map.
+	//   If the whole API is required, then mark it in the `apiGroupVersionRanges` map.
```

Comment:
Maybe rather link the MD document with the explanation instead of duplicating it here?